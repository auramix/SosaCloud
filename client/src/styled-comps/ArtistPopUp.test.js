import React from 'react';
import expect from 'expect';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {
  ArtistPopUp,
  PopUpDiv,
  ArtistAnchor
} from './ArtistPopUp.jsx';

describe('ArtistPopUp', () => {
  it('Has correct number of children', () => {
    const wrapper = shallow( < ArtistPopUp / > , {
      disableLifecycleMethods: true
    });

    expect(wrapper.find('.artist-pop-up').children('div')).toHaveLength(1);
    expect(wrapper.find('.artist-pop-up').children('div').children('*')).toHaveLength(5);
  });

  // Snapshot test
  it('Has expected snapshot output', () => {
    const wrapper = shallow( <
      PopUpDiv artistName = {
        "testname"
      }
      />
    );
    expect(wrapper).toMatchSnapshot();
  })

});

describe('PopUpDiv', () => {
      it('Has a visibility property', () => {
          const wrapper = renderer.create( < PopUpDiv artistName = {
              "testname"
            }
            />).toJSON()
            expect(wrapper).toHaveStyleRule('visibility', 'hidden');
          });
      });