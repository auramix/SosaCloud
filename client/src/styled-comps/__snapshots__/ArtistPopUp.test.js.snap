// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistPopUp Has expected snapshot output 1`] = `
.c0 {
  visibility: hidden;
  position: absolute;
  background-color: #fff;
  width: auto;
  min-width: 135px;
  min-height: 199px;
  border-radius: 4px;
  padding: 10px;
  border: 1px solid #ccc;
  box-shadow: 0 2px 10px rgba(0,0,0,.1);
  text-align: center;
  z-index: 1;
}

<div
  className="c0"
/>
`;
